src/components/AlbumsList.js

```diff
@@ -56,6 +58,29 @@ const AlbumsList = () => {
     fetchAlbums('albumName', '', 10);
   }, []);
 
+  //fetch wishlist album from API
+  useEffect(() => {
+    const fetchWishlist = async () => {
+      try {
+        const response = await axiosInstance.post(`/wishlist/`); //use axiosInstance to send cookie token with request
+        const wishlistData = response.data.wishlist;
+        setWishListId(wishlistData._id);
+        setWishListAlbums(wishlistData.albums);
+        // Store each album id from wishlist in local storage
+        wishlistData.albums.forEach((album) => {
+          localStorage.setItem(
+            `wishListAlbums-${album._id}`,
+            JSON.stringify(true)
+          );
+        });
```

Comment:
This is fine (and I know initially this is the same pattern the `addToWishlistButton` was using) but eventually if wishlist is large, we'll end up with a lot of entries in localstorage which may get difficult to manage or even cause browser slowdowns.

Let's try to store it in localstorage as an object:

```suggestion
        // Store object in local storage where keys are album id and value is the whole album
        const wishlistAlbumsToStore = wishlistData.albums.reduce((acc, album) => {
          acc[album._id] = album;
          return acc;
        }, {});
        localStorage.setItem("wishlistAlbums", JSON.stringify(wishlistAlbumsToStore));
```

or if you prefer to do it iteratively  (not with `Array.reduce`):

```suggestion
        // Store object in local storage where keys are album id and value is the whole album
        const wishlistAlbumsToStore = {};
        wishlistData.albums.forEach((album) => {
          wishlistAlbumsToStore[album._id] = album
        });
        localStorage.setItem("wishlistAlbums", JSON.stringify(wishlistAlbumsToStore));
```

Reply:
I changed it to object